test/e2e/cni/helpers.go

```diff
@@ -0,0 +1,445 @@
+// TODO figure out if this name should change or where these functions should go
+package cni
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/ipamd/datastore"
+	"github.com/tiffanyfay/aws-k8s-test-framework/test/e2e/framework"
+	"github.com/tiffanyfay/aws-k8s-test-framework/test/e2e/resources"
+
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/awsutils"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/cihub/seelog"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// UpdateAWSNodeEnvs updates the aws-node daemonset's WARM_IP_TARGET, WARM_ENI_TARGET, and MAX_ENI and waits for
+// the daemonset to be updated
+// TODO make an env replace function that is more generic
+func UpdateAWSNodeEnvs(ctx context.Context, f *framework.Framework, warmIPTarget, warmENITarget, maxENI string) {
```

Comment:
This method could probably take a `map[string]string` instead, then just apply whatever is passed in as long as it matches, or add if it's missing.

Reply:
Agreed! I just made a new function yesterday after this to do that.